src/main/java/com/example/application/config/AutowiredConfig.java

```diff
@@ -0,0 +1,41 @@
+package com.example.application.config;
+
+import com.example.application.dao.ClientDAO;
+import com.example.application.dao.ContractDAO;
+import com.example.application.dao.OptionsDAO;
+import com.example.application.util.EntityManagerUtil;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+@Configuration
+
+
+public class AutowiredConfig {
+
+    @Bean
+    public static EntityManagerFactory getEntityManagerFactory() {
+        return EntityManagerUtil.getEntityManagerFactory();
+    }
+
+    @Bean
+    public static EntityManager getEntityManager() {
+        return EntityManagerUtil.getEntityManagerFactory().createEntityManager();
+    }
+
+    @Bean
+    public static ClientDAO getClientDAO() {
```

Comment:
You make annotate DAOs with Component. You don't need this. Also, for all such beans